lib/galaxy/objectstore/irods.py

```diff
@@ -454,20 +454,20 @@ def file_ready(self, obj, **kwargs):
         return False
 
     def _exists(self, obj, **kwargs):
-        in_cache = in_irods = False
+        in_irods = False
```

Comment:
You can simplify this method further. The github interface doesn't let me create a nice multi-line suggestion here, but this is what it boils down to:
```python
if self._in_cache(rel_path) or self._data_object_exists(rel_path):
    return True
```
At this point the only way we can still return `True` is if `dir_only ` and `base_dir` are both not "False-y".
So you can do
```python
if dir_only and base_dir:
    if not os.path.exists(rel_path):
        os.makedirs(rel_path)
    return True
return False
```

Reply:
Good idea. Revised code accordingly.